crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -279,18 +279,44 @@ async fn app_validation_workflow_inner(
 
                 #[cfg(feature = "unstable-warrants")]
                 if let Outcome::Rejected(_) = &outcome {
-                    let keystore = conductor.keystore();
-                    let warrant_op =
-                        crate::core::workflow::sys_validation_workflow::make_invalid_chain_warrant_op(
-                            keystore,
-                            _representative_agent.clone(),
-                            &chain_op,
-                        )
-                        .await?;
+                    let mut issue_warrant = true;
+                    match holochain_state::warrant::is_action_warranted_as_invalid(
+                        &workspace.dht_db,
+                        chain_op.action().to_hash(),
+                        chain_op.author().clone(),
+                    )
+                    .await
+                    {
+                        Ok(true) => {
+                            tracing::trace!(
+                                "Op {} is already warranted, not issuing a new warrant",
+                                dht_op_hash
+                            );
+                            issue_warrant = false;
+                        }
+                        Ok(false) => {
+                            // Not warranted yet, should issue a warrant.
+                        }
+                        Err(e) => {
+                            tracing::error!(error = ?e, "Error checking if op is warranted");
+                            issue_warrant = false;
+                        }
+                    }
```

Comment:
Take it or leave it =)
```suggestion
                    let issue_warrant = match holochain_state::warrant::is_action_warranted_as_invalid(
                        &workspace.dht_db,
                        chain_op.action().to_hash(),
                        chain_op.author().clone(),
                    )
                    .await
                    {
                        Ok(true) => {
                            tracing::trace!(
                                "Op {} is already warranted, not issuing a new warrant",
                                dht_op_hash
                            );
                            false
                        }
                        Ok(false) => {
                            // Not warranted yet, should issue a warrant.
                            true
                        }
                        Err(e) => {
                            tracing::error!(error = ?e, "Error checking if op is warranted");
                            false
                        }
                    }
```